Add TUNNEL_MODE_STRING and LB mode to start.sh

mchalla opened the pull request:
LB mode
This would enable multi node config using load balancer with direct server
return. The even numbered nodes are clients and odd numbered servers. It
affects the routes generated so the clients reach servers via load balancer
and server reaches clients directly. cilium-master will be configured as
load balancer.

TUNNEL_MODE_STRING
Tells whether or not to use a tunnel. Default is "-t vxlan"

Signed-off-by: Madhu Challa madhu@cilium.io

aanm commented on contrib/vagrant/start.sh:
```diff
@@ -49,12 +53,22 @@ EOF
         hexIPv6=$(printf "%04X" "${index}")
-        cat <<EOF >> "${filename}"
+
+	if [ "$LB" = 1 ] && [ $((node_index%2)) -eq 0 ]; then
```
@mchalla Can you explain me what is this line for?

mchalla replied:
@aanm: It means if we are in load balancer mode $LB=1 and an even node %2 we are a client node and we set the route index to point to the load balancer node. For odd numbered nodes the route index will point to the actual node just like when LB=0, i.e the behavior today.

aanm replied:
@mchalla okay, please add a small comment regarding that before the `if`

mchalla committed:
Add TUNNEL_MODE_STRING and LB mode to start.sh
LB mode
This would enable multi node config using load balancer with direct server
return. The even numbered nodes are clients and odd numbered servers. It
affects the routes generated so the clients reach servers via load balancer
and server reaches clients directly. cilium-master will be configured as
load balancer.

TUNNEL_MODE_STRING
Tells whether or not to use a tunnel. Default is "-t vxlan"

Signed-off-by: Madhu Challa <madhu@cilium.io>

Add a comment about client and server nodes for load balancer.

Signed-off-by: Madhu Challa <madhu@cilium.io>
Acked-by: André Martins <andre@cilium.io>
Acked-by: Thomas Graf <thomas@cilium.io>
